app/controllers/exercises_controller.rb

```diff
@@ -25,16 +25,19 @@ def create
     end
   end
 
-  def mark_as_finished
-    user_exercise = UserExercise.find_by(user_id: current_user.id, exercise_id: params[:exercise_id])
-    user_exercise.update_attribute(:status, "finished")
-    render nothing: true
+  def submit
+    @exercise = Exercise.find(params[:exercise_id]);
+    @submitResult = SubmitResult.new(params[:answer], @exercise.keywords_hash)
+    if (@submitResult.is_successfull())
```

Comment:
in Ruby a function usually doesn't need empty paranthesis at the end, I would remove those here and from the method definition in the class